pkg/datasource/csvsource/stream.go

```diff
@@ -0,0 +1,158 @@
+package csvsource
+
+import (
+	"encoding/csv"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+//go:generate callbackgen -type Stream
+type Stream struct {
+	types.StandardStream
+	config                    *CsvStreamConfig
+	converter                 ICSVTickConverter
+	marketTradeEventCallbacks []func(e []CsvTick)
+	kLineEventCallbacks       []func(e []types.KLine)
+	orderEventCallbacks       []func(e []types.Order)
+	tradeEventCallbacks       []func(e []types.Trade)
+}
+
+type CsvStreamConfig struct {
+	Interval     types.Interval     `json:"interval"`
+	RateLimit    time.Duration      `json:"rateLimit"`
+	StrategyID   string             `json:"strategyID"`
+	CsvPath      string             `json:"csvPath"`
+	Exchange     types.ExchangeName `json:"exchange"`
+	Symbol       string             `json:"symbol"`
+	BaseCoin     string             `json:"baseCoin"`
+	QuoteCoin    string             `json:"quoteCoin"`
+	TakerFeeRate fixedpoint.Value   `json:"takerFeeRate"`
+	MakerFeeRate fixedpoint.Value   `json:"makerFeeRate"`
+}
+
+func NewStream(cfg *CsvStreamConfig) *Stream {
```

Comment:
this Stream seems not completed yet, could you remove this or move this to another new PR?

Reply:
You are right I just wanted to get a little feedback. I will put it into a separate pr.